fv3core/fv3core/_config.py

```diff
@@ -265,10 +265,10 @@ class DynamicalCoreConfig:
     nf_omega: int = NamelistDefaults.nf_omega
     fv_sg_adj: int = NamelistDefaults.fv_sg_adj
     n_sponge: int = NamelistDefaults.n_sponge
-    namelist_override: str = None
+    namelist_override: Optional[str] = None
 
     def __post_init__(self):
-        if self.namelist_override is not None:
+        if self.namelist_override:
```

Comment:
nit: it's a bit clearer to read this how it was - checking the boolean evaluation of it makes me think namelist_override is a boolean switch. It's more relevant for this than some variables since namelist_override also sounds more like it could be a boolean than something like e.g. `override_path`.

```suggestion
        if self.namelist_override is not None:
```